src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends object>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  return items.reduce((acc, item) => {
+    const value = item[key];
+
+    if (typeof value === 'string' || typeof value === 'number') {
+      if (!acc[value]) {
+        acc[value] = [];
+      }
+
+      acc[value].push(item);
+    }
+
+    return acc;
```

Comment:
Using 'reduce' is an excellent choice for building up the GroupsMap. However, it might be beneficial to declare the initial value of the accumulator outside of the reduce call for better readability. For example, 'const initialAcc: GroupsMap<T> = {};' and then use 'initialAcc' as the second argument to 'reduce'.